packages/common/src/velocityHelpers.ts

```diff
@@ -0,0 +1,130 @@
+/** @license
+ * Copyright 2021 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { UserSession } from "./interfaces";
+import {
+  getSubscriptionInfo,
+  ISubscriptionInfo
+} from "./get-subscription-info";
+import { getProp } from "./generalHelpers";
+
+/**
+ * Get the base velocity url from the current orgs subscription info
+ *
+ * This function will update the input templateDictionary arg with the velocity url
+ * so we can reuse it without pinging the org again for subsequent requests to the
+ * velocity api.
+ *
+ * @param authentication Credentials for the requests
+ * @param templateDictionary Hash of facts: folder id, org URL, adlib replacements
+ *
+ * @return a promise that will resolve with the velocity url
+ *
+ */
+export function getVelocityUrlBase(
+  authentication: UserSession,
+  templateDictionary: any
+): Promise<string> {
+  // if we already have the base url no need to make any additional requests
+  if (templateDictionary.velocityUrl) {
+    return Promise.resolve(templateDictionary.velocityUrl);
+  } else {
+    // get the url from the orgs subscription info
+    return getSubscriptionInfo({ authentication }).then(
+      (subscriptionInfo: ISubscriptionInfo) => {
+        let velocityUrl = "";
+        const orgCapabilities = getProp(subscriptionInfo, "orgCapabilities");
+        /* istanbul ignore else */
+        if (Array.isArray(orgCapabilities)) {
+          orgCapabilities.some(c => {
+            velocityUrl = c.velocityUrl;
+            return velocityUrl;
+          });
+        }
+        // add the base url to the templateDictionary for reuse
+        templateDictionary.velocityUrl = velocityUrl;
```

Comment:
Do you intend to set `templateDictionary.velocityUrl` to "" in the case where "orgCapabilities" is not returned?

Reply:
@MikeTschudi good catch...the "" will be ok but I do need to make some changes to properly avoid redundant and bad requests when the org doesn't support velocity.

Reply:
@MikeTschudi we will prevent users from deploying solutions that have velocity items via an upcoming change to the app but I didn't handle it correctly for users of solution.js outside the app